src/status_im2/contexts/onboarding/select_photo/method_menu/view.cljs

```diff
@@ -45,9 +46,24 @@
 (defn pick-pic
   [update-profile-pic-callback]
   (rf/dispatch [:hide-bottom-sheet])
-  (show-image-picker
-   #(update-profile-pic-callback (.-path ^js %))
-   crop-opts))
+
+  (permissions/permission-granted?
+   :read-external-storage
+   (fn [granted]
+     (if granted
+       (show-image-picker
+        #(update-profile-pic-callback (.-path ^js %))
+        crop-opts)
+       (permissions/request-permissions
+        {:permissions [:read-external-storage]
+         :on-allowed  (fn []
```

Comment:
This seems like it should be generic code and 'on-allowed' should be a parameter of this whole function. We have many places where the camera is, eg chat. in the onboarding flows it is not mandatory to add a photo so there is no guarantee the user would see the camera permissions there.